service/controller/machinepool/handler/migration/machinepool_test.go

```diff
@@ -0,0 +1,294 @@
+package migration
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	oldcapiexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capiexp/v1alpha3"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+
+	"github.com/giantswarm/azure-operator/v5/service/unittest"
+)
+
+func TestEnsureCreatedMachinePoolIsCorrect(t *testing.T) {
+	testCases := []struct {
+		name               string
+		oldMachinePoolFile string
+		newMachinePoolFile string
+		expected           expected
+	}{
+		{
+			name:               "case 0: New MachinePool references not updated",
+			oldMachinePoolFile: "old_machinepool.yaml",
+			newMachinePoolFile: "new_machinepool_references_not_updated.yaml",
+		},
+		{
+			name:               "case 1: New MachinePool status not updated",
+			oldMachinePoolFile: "old_machinepool.yaml",
+			newMachinePoolFile: "new_machinepool_status_not_updated.yaml",
+		},
+		{
+			name:               "case 2: New MachinePool fully created",
+			oldMachinePoolFile: "old_machinepool.yaml",
+			newMachinePoolFile: "new_machinepool_fully_created.yaml",
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			t.Log(tc.name)
+			ctx := context.Background()
+
+			// Load old MachinePool
+			oldMachinePool, err := loadOldMachinePoolCR(tc.oldMachinePoolFile)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Load new MachinePool, the one which would be created by machinepoolexp/migration handler
+			newMachinePool, err := loadMachinePoolCR(tc.newMachinePoolFile)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			fakeK8sClient := unittest.FakeK8sClient(oldMachinePool, newMachinePool)
+
+			// Create machinepool/migration handler
+			var resource *Resource
+			{
+				c := Config{
+					CtrlClient: fakeK8sClient.CtrlClient(),
+					Logger:     microloggertest.New(),
+				}
+				resource, err = New(c)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+
+			// Now run EnsureCreate for the new MachinePool
+			err = resource.EnsureCreated(ctx, newMachinePool)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Assert new MachinePool is correct
+			namespacedName := types.NamespacedName{
+				Namespace: newMachinePool.Namespace,
+				Name:      newMachinePool.Name,
+			}
+			newMachinePoolAfterEnsureCreated := &capiexp.MachinePool{}
+			verifyCR(t, fakeK8sClient.CtrlClient(), tc.name, newMachinePoolAfterEnsureCreated, namespacedName)
+		})
+	}
+}
+
+func TestEnsureCreatedMachinePoolUpdated(t *testing.T) {
```

Comment:
Test that checks if the expected number of new MachinePool CR updates happened in the EnsureCreated func (in a way testing that the func is idempotent).